File: programing-language/内置函数/test_abs.py
a = 1
print(a, type(a), abs(a))
b = -1.0
print(b, type(b), abs(b))
c = 2+0j
print(c, type(c), abs(c))
class X:
    def __init__(self, _x):
        self._x = _x

    def __abs__(self) :
        if self._x < 0 : return -self._x
        return self._x


d = X(-2)
print(d, type(d), abs(d))

File: programing-language/内置函数/test_abs.java
public class test_abs{
    public static void main(String args[]){
        Integer a = -1;
        double b = 2;
        long c = -9223372036854775808L;
        long d = -9223372036854775807L;
        System.out.println(Math.abs(a));
        System.out.println(Math.abs(b));
        System.out.println(Math.abs(c));
        System.out.println(Math.abs(d));
    }
}